File: app/src/main/java/dev/voleum/ordermolder/viewmodels/GoodsOrderListItemViewModel.java
package dev.voleum.ordermolder.viewmodels;

import android.view.View;

import androidx.databinding.Bindable;

import dev.voleum.ordermolder.R;
import dev.voleum.ordermolder.models.TableGoods;

public class GoodsOrderListItemViewModel extends AbstractDocListItemViewModel<TableGoods, OrderViewModel> implements View.OnClickListener {

    public GoodsOrderListItemViewModel(TableGoods table, OrderViewModel viewModel) {
        super(table, viewModel);
    }

    @Bindable
    public void setQuantity(String quantity) {
        double q = 0.0;
        try {
            q = Double.parseDouble(quantity);
        } catch (NumberFormatException ignored) {

        }
        getTable().setQuantity(q);
        getTable().countSum();
        getViewModel().countSum();
    }

    @Bindable
    public String getQuantity() {
        return String.valueOf(getDf().format(getTable().getQuantity()));
    }

    @Bindable
    public void setPrice(String price) {
        double p = 0.0;
        try {
            p = Double.parseDouble(price);
        } catch (NumberFormatException ignored) {

        }
        getTable().setPrice(p);
        getTable().countSum();
        getViewModel().countSum();
    }

    @Bindable
    public String getPrice() {
        return String.valueOf(getDf().format(getTable().getPrice()));
    }

    @Override
    public void onClick(View v) {
        switch (v.getId()) {
            case R.id.good_plus:
                getTable().increaseQuantity();
                break;
            case R.id.good_minus:
                getTable().decreaseQuantity();
                break;
        }
        notifyChange();
        getViewModel().countSum();
    }
}

File: app/src/main/java/dev/voleum/ordermolder/viewmodels/GoodsOrderListItemViewModel.kt
package dev.voleum.ordermolder.viewmodels

import android.view.View
import androidx.databinding.Bindable
import dev.voleum.ordermolder.R
import dev.voleum.ordermolder.models.TableGoods

class GoodsOrderListItemViewModel(table: TableGoods, viewModel: OrderViewModel) : AbstractDocListItemViewModel<TableGoods, OrderViewModel>(table, viewModel), View.OnClickListener {

    var quantity: String = ""
        @Bindable set(value) {
            val q = if (value == "") 0.00 else value.toDouble()
            table.quantity = q
            table.countSum()
            viewModel.countSum()
            field = q.toString()
        }
        @Bindable get() = df.format(table.quantity)

    var price: String = ""
        @Bindable set(value) {
            val p = if (value == "") 0.00 else value.toDouble()
            table.price = p
            table.countSum()
            viewModel.countSum()
            field = p.toString()
        }
        @Bindable get() = df.format(table.price)

    override fun onClick(v: View?) {
        when (v?.id) {
            R.id.text_input_start_icon -> table.decreaseQuantity()
            R.id.text_input_end_icon -> table.increaseQuantity()
//            R.id.good_plus -> table.increaseQuantity()
//            R.id.good_minus -> table.decreaseQuantity()
        }
        notifyChange()
    }
}